### ISSUE TITLE
Longtime token with connectToSpotifyRemote

### ISSUE THREAD

**Issue by PaulDomixApp** [brim-borium/spotify_sdk#23]
The remote spotify connection time out after around 10 minutes so the user need to reconnect to the spotify remote.

1. connect to spotify remote
2. launch the spotify player
3. play a playlist
4. The spotify remote is killed after 10 minutes

I want a longtime token to avoid reconnection to the spotify remote 

On every devices i tested

**Comment by irfanbacker** [comment]
I have the same problem. The connection is automatically disconnected after sometime

**Comment by fotiDim** [comment]
Not sure how it is supposed to be on Android but on iOS this is standard behavior. When the music is paused then the connection is dropped and you have to reconnect again. Reconnection though is transparent to the user so in general is not a problem.

If disconnection events are not forwarded to the plugin then we need to look into that. For example what happens if you kill the Spotify app while your app is connected?

**Comment by irfanbacker** [comment]
> If disconnection events are not forwarded to the plugin then we need to look into that. For example what happens if you kill the Spotify app while your app is connected?

There is nothing in the API which we can use to check the connection. The API only provides a connectToSpotify function. It would be really good if that is added.

**Comment by Joran-Dob** [comment]
Will start investigating the desired behaviour in this case for the package. 

It might be nice to inform the developer that the connection dropped.

**Comment by Joran-Dob** [comment]
@irfanbacker Could you provide more information about then the connection is blocked on pause as when testing on Android when pausing the playback in the app, in the remote Spotify app and Spotify desktop. I cannot get the connection to drop here. 

@PaulDomixApp tried to have the app open longer than 10 minutes and it seems the app example app stays connected to the Spotify remote for 60 minutes also as specified [here](https://developer.spotify.com/documentation/general/guides/authorization-guide/), so you want the save token as proposed here #55  so you can do a server-side token refresh for a longtime token?

**Comment by ollydixon** [comment]
We do the refresh and access tokens server side so connection doesn't matter. They always have the latest access token no matter what, it works flawlessly but our current limitation is we need to use a WebView to get the refresh token and access token.

Example code
```
// // Setup launch
    final credentials = SpotifyApiCredentials(clientId, clientSecret);
    final grant = SpotifyApi.authorizationCodeGrant(credentials);
    final scopes = [
      'streaming',
      'app-remote-control',
      'playlist-read-collaborative',
      'playlist-read-private',
      'user-read-playback-state',
      'user-read-recently-played',
    ];
    final authUri = grant.getAuthorizationUrl(
      Uri.parse(redirectUri),
      scopes: scopes,
    );

    // Listen to the webview
    final flutterWebviewPlugin = new FlutterWebviewPlugin();
    flutterWebviewPlugin.onUrlChanged.listen((String url) async {
      if (url.startsWith('jukebox-url-scheme://spotify-login-callback')) {
        if (Navigator.canPop(context)) {
          Navigator.pop(context);
        }
        final spotify = SpotifyApi.fromAuthCodeGrant(grant, url);
        final creds = await spotify.getCredentials();
        await request('spotify-add-auth-token', data: {
          'accessToken': creds.accessToken,
          'refreshToken': creds.refreshToken,
          'tokenExpiresISO': creds.expiration.toUtc().toIso8601String(),
        });
        await fetchUser();

        await SpotifySdk.connectToSpotifyRemote(
          clientId: clientId,
          redirectUrl: "jukebox-url-scheme://spotify-login-callback",
        );
      }
    });

    // Open a webview container
    Navigator.push(
      context,
      slideScreenFromBottom(
        WebviewScaffold(
          url: authUri.toString(),
          withJavascript: true,
          withLocalStorage: true,
          appBar: AppBar(
            elevation: 0,
            backgroundColor: Colors.black,
            title: Text(getLangauge().allowSpotify),
          ),
        ),
      ),
    );
```

I logged an issue here with an example, there's an example here of the node code.
https://github.com/brim-borium/spotify_sdk/issues/55

At https://developer.spotify.com/documentation/general/guides/authorization-guide/
Specially we need "Authorization Code With PKCE" to work with the native connect.

So this library needs to support authorisation with PKCE for it to be perfect :-)

**Comment by fotiDim** [comment]
I am reading 2 different things in this issue:
>     4. The spotify remote is killed after 10 minutes
Disconnections after music is paused or when Spotify is killed are normal. Use the `subscribeConnectionStatus()` stream to react on them.
Disconnections while playing music did not happen during out testing on either platform. Download the latest release and if you are still experiencing this please open a new issue.

> I want a longtime token to avoid reconnection to the spotify remote

Spotify access tokens are meant to last for 1 hour. There is no way to get a refresh token from the native Android SDK and consequently not from this SDK. For potential updates follow [this](https://github.com/spotify/android-sdk/issues/255) and [this](https://github.com/spotify/android-sdk/issues/12) issue.
